### PR TITLE
Rename hip/hip_hcc.h to hip/hip_ext.h

### PR DESCRIPTION
**Pull request by gargrahul**
With these changes #1338 shouldn't be needed.

### CODE REVIEW

**Comment by gargrahul** [comment]
For consistency purposes I plan to move all hipExtXXX APIs in hip_ext.h.

**Comment by yxsamliu** [comment]
hip_ext.h needs to be included in hip_runtime.h for hip-clang. Currently it is not. It is included in the grid launch header files, but those headers are included for HCC only. hip-clang cannot includes those grid launching header files since they conflicts with hip-clang's kernel launching mechanism. Also hip_ext contains some other HCC API functions which should be disabled for hip-clang.

**Comment by yxsamliu** [comment]
I think hip_ext.h needs to be moved from grid launching headers to hip_runtime.h

**Comment by mhbliao** [comment]
better to split hip_ext.h into a general part shared by HIP-clang and HCC and included in hip_runtime.h. The remaining HCC specific ones  (hip_hcc.h or similar naming) still remain the same.

> I think hip_ext.h needs to be moved from grid launching headers to hip_runtime.h

File: include/hip/hip_ext.h [hunk 1]
```diff
@@ -22,10 +22,8 @@ THE SOFTWARE.
 
 #ifndef HIP_INCLUDE_HIP_HIP_HCC_H
 #define HIP_INCLUDE_HIP_HIP_HCC_H
-
-#ifdef __HCC__
-
 #include "hip/hip_runtime_api.h"
+#ifdef __HCC__
```
Comment (aaronenyeshi) [inline include/hip/hip_ext.h]: Please include this header, and also use #ifdef __HCC__ && __HIP_VDI__ for HIP-Clang support.

File: include/hip/hip_ext.h [hunk 2]
```diff
@@ -22,10 +22,8 @@ THE SOFTWARE.
 
 #ifndef HIP_INCLUDE_HIP_HIP_HCC_H
```
[the remark this answers is not in the document]
Comment (scchan) [reply include/hip/hip_ext.h]: @gargrahul Renaming this header into hip_ext.h may break some codes out there.  Can we create a new header called hip_ext.h and then change hip_hcc.h to include hip_ext.h?  hip_hcc.h should be marked deprecated

File: include/hip/hip_ext.h (hunk 1 above)
Comment (scchan) [reply include/hip/hip_ext.h]: but hip-clang doesn't `hc::accelerator` and `hc::accelerator_view`

File: include/hip/hip_ext.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (mangupta) [reply include/hip/hip_ext.h]: Agree with @scchan here. We cannot rename public header filenames. Solution suggested above seems to be acceptable. It does not break any existing application and also gives them enough warning to migrate to directly including hip_ext.h.

**Pull request merged**